pkg/sidecar/param/pub_api.go

```diff
@@ -0,0 +1,238 @@
+// Copyright © 2018 One Concern
+package param
+
+import (
+	"errors"
+)
+
+type fuseParamsBundleParams struct {
+	Name         string `json:"name" yaml:"name"`
+	SrcPath      string `json:"srcPath" yaml:"srcPath"`
+	SrcRepo      string `json:"srcRepo" yaml:"srcRepo"`
+	SrcLabel     string `json:"srcLabel" yaml:"srcLabel"`
+	SrcBundle    string `json:"srcBundle" yaml:"srcBundle"`
+	DestPath     string `json:"destPath" yaml:"destPath"`
+	DestRepo     string `json:"destRepo" yaml:"destRepo"`
+	DestMessage  string `json:"destMessage" yaml:"destMessage"`
+	DestLabel    string `json:"destLabel" yaml:"destLabel"`
+	DestBundleID string `json:"destBundleID" yaml:"destBundleID"`
```

Comment:
is `DestBundleID` a parameter passed by the client?

Reply:
could be better named ... this is a path to a file where the bundle id is written after upload.

the idea of such a specification was an as-needed addition when supporting the seismic pipeline integration with @amarkovits-1C ..
.. i recall, at the time, having some more abstracted plan of implementation that could then be retrofitted to the specific requirements of the seismic workflow.  i think it was something along the lines of having a unified artifact according to some format describing the results of the sidecar, just as we're now developing such an object for the parameterization?  i.e. totally RPC-ify sidecar usage

we implemented the simplest possible thing that fulfilled the seismic workflows requirements with the bundle id file to avoid a separate post-processing step.  and here i've again done the simple thing by preserving the extant interface, re-shaping it according to the parameterization rather than `optargs` as it was originally implemented.

if it's viewed as more useful to have a more general [un]marshallable output format defined in much the same way as the parameterization format is here described, i could implement that as well or instead of this parameter?

Reply:
Got it, I was a bit confused by `DestBundleID` because it seemed to be the bundleID and not a file path. It's not really required, but calling it DestBundleIDFile or something similar might make it more clear.